src/base/utils/misc.cpp

```diff
@@ -86,6 +86,31 @@ namespace
     };
 }
 
+#if __cplusplus < 201703L
+namespace
+{
+    // tests not in std:: namespace to minimize possible clatter
+    template <class T> struct is_duration : std::false_type {};
+    template <class Rep, class Period> struct is_duration<
+        std::chrono::duration<Rep, Period>> : std::true_type {};
+}
+namespace std
+{
+namespace chrono
+{
+    template <class To, class Rep, class Period,
+        class = std::enable_if_t<is_duration<To>{}>>
+    constexpr To floor(const duration<Rep, Period>& d)
```

Comment:
you're not actually allowed to put things in the `std` namespace. I think a better way to do this is to put this template in your namespace and when building with C++17, just pull it in from `std` with `using` instead.

Reply:
Seems like we do not use negative durations, thus `duration_cast` is enough, there is no need tor `floor`.